pkg/operator/ceph/cluster/mon/util.go

```diff
@@ -42,21 +40,16 @@ func monInQuorum(monitor client.MonMapEntry, quorum []int) bool {
 
 // convert the mon name to the numeric mon ID
 func fullNameToIndex(name string) (int, error) {
-	prefix := AppName + "-"
-	if strings.Index(name, prefix) != -1 && len(prefix) < len(name) {
-		return k8sutil.NameToIndex(name[len(prefix)+1:])
+	// remove the "rook-ceph-mon" prefix
+	if strings.HasPrefix(name, AppName) {
+		name = name[len(AppName):]
 	}
-
-	// attempt to parse the legacy mon name
-	legacyPrefix := AppName
-	if strings.Index(name, legacyPrefix) == -1 || len(name) < len(AppName) {
-		return -1, errors.New("unexpected mon name")
-	}
-	id, err := strconv.Atoi(name[len(legacyPrefix):])
-	if err != nil {
-		return -1, err
+	// remove the "-" prefix
+	if strings.HasPrefix(name, "-") {
```

Comment:
Can't we just remove `rook-ceph-mon-` in one command?

Reply:
There are some legacy mon names in slightly different formats. Handling the `-` separately was done before so it is safer to preserve that behavior.